CAIPs/caip-50.md

```diff
@@ -0,0 +1,207 @@
+---
+caip: 50
+title: Multi-Chain Account ID Specification
+author: Joel Torstensson (@oed), Pedro Gomes (@pedrouid)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/pull/50
+status: Draft
+type: Standard
+created: 2020-06-10
+updated: 2020-06-11
+requires: 2
+---
+
+## Simple Summary
+
+CAIP-50 defines a way to identify blockchain account addresses uniquely across multiple blockchains
+
+## Abstract
+
+This proposal aims to facilitate using unique address for accounts on multiple blockchain systems using a multi-codec format that encodes variable integer for chain identifiers complaint with CAIP-2 blockchain id specification. This is useful for both decentralized applications and wallets to communicate user accounts for multiple chains using unique identifiers which are machine veriable and can be decoded to identify the original address and chainId encoded. This proposal aims to standardize these identifiers for accounts to allow inteoperability for multi-chain applications.
+
+## Motivation
+
+The motivation for this proposal came from different feedback received from the adoption of CAIP-10 and tackles these to provide a significant improvement in more efficient communication of these identifiers using smaller byte footprint while preserving uniqueness and interoperability.
+
+## Specification
+
+In this specifiation we define the Multi-Chain Account Id or MACI for short which is a compactly encoded account identifier that is contextual to a blockchain using a CAIP-2 chainId.
+
+### Syntax
+
+```js
+mcai ::= <multibase_prefix><mcai_code><chain_namespace><id_size><chain_id><address_size><address><parity_byte>
+```
+
+### Semantics
+
+- `multibase_prefix` - the prefix which defines which multibase is used to encode the bytes, `z` for `base58btc`
+- `mcai_code` - a number registered on the [multicodec table](https://github.com/multiformats/multicodec/blob/master/table.csv), makes the multi-chain account id upgradable, encoded as varint
+- `chain_namespace` - see table below, encoded as varint
+- `id_size` - the length in bytes of the `chain_id`, encoded as varint
+- `chain_id` - the chain id, encoding is defined by the chain namespace
+- `address_size` - the length of the address, encoded as varint
+- `address` - the address itself, encoding is defined by the chain namespace
+- `parity_byte` - a checksum byte, see section below
+
+### MCAI multicodec
+
+Should be a number registered on the [multicodec table](https://github.com/multiformats/multicodec/blob/master/table.csv). In the examples below we use `0xCA` but this is subject to change.
+
+### Chain namespaces
+
+Each blockchain namespace needs to be properly defined with a registry table:
+
+#### Registry Table
+
+| Namespace | code |
+| --------- | ---- |
+| bip122    | 0x00 |
+| eip155    | 0x01 |
+| cosmos    | 0x02 |
+| polkadot  | 0x03 |
```

Comment:
There could be more than one relay chains in Polkadot 2.0.
Right now there are widely recognized -- Polkadot & Kusama.

Reply:
So should there be one chain_namespace for each relay chain?

Reply:
I think that technically the polkadot namespace would need to be renamed as "substrate" instead because it we already specify Polkadot and Kusama with different chainId's for CAIP-23

```
# Polkadot
polkadot:91b171bb158e2d3848fa23a9f1c25182

# Kusama
polkadot:b0a8d493285c2df73290dfb7e61f870f
```

Reply:
I guess, going with **dot** would also work for a while as recognition for ecosystem, where substrate might long for some, still someone could fork substrate as well :)